File: src/main/java/no/nav/sbl/service/VeilederService.java
package no.nav.sbl.service;

import no.nav.sbl.rest.domain.DecoratorDomain;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.cache.annotation.Cacheable;

import java.util.Map;

import static java.util.Arrays.asList;

public class VeilederService {

    @Autowired
    LdapService ldapService;


    @Cacheable("veilederCache")
    public DecoratorDomain.Saksbehandler hentVeilederNavn(String ident) {
        Map map = ldapService.hentVeilederAttributter(ident, asList("givenname", "sn"));

        return new DecoratorDomain.Saksbehandler(
                ident,
                map.get("givenname").toString(),
                map.get("sn").toString()
        );
    }
}

File: src/main/java/no/nav/sbl/service/VeilederService.kt
package no.nav.sbl.service;

import no.nav.common.client.nom.NomClient;
import no.nav.common.client.nom.VeilederNavn;
import no.nav.common.types.identer.NavIdent;
import no.nav.sbl.rest.domain.DecoratorDomain;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.cache.annotation.Cacheable;

public class VeilederService {

    @Autowired
    NomClient nomClient;

    @Cacheable("veilederCache")
    public DecoratorDomain.Saksbehandler hentVeilederNavn(String ident) {
        VeilederNavn veilederNavn = nomClient.finnNavn(new NavIdent(ident));

        return new DecoratorDomain.Saksbehandler(
                ident,
                veilederNavn.getFornavn(),
                veilederNavn.getEtternavn()
        );
    }
}
